rstar/src/point.rs

```diff
@@ -344,3 +346,90 @@ implement_point_for_array!(0, 1, 2, 3, 4, 5);
 implement_point_for_array!(0, 1, 2, 3, 4, 5, 6);
 implement_point_for_array!(0, 1, 2, 3, 4, 5, 6, 7);
 implement_point_for_array!(0, 1, 2, 3, 4, 5, 6, 7, 8);
+
+macro_rules! fixed_type {
+    ($expr:expr, $type:ty) => {
+        $type
+    };
+}
+
+macro_rules! impl_point_for_tuple {
+    ($($index:expr => $name:ident),+) => {
+        impl<S> Point for ($(fixed_type!($index, S),)+)
+        where
+            S: RTreeNum
+        {
+            type Scalar = S;
+
+            const DIMENSIONS: usize = count_exprs!($($index),*);
+
+            fn generate(generator: impl Fn(usize) -> S) -> Self {
+                ($(generator($index),)+)
+            }
+
+            #[inline]
+            fn nth(&self, index: usize) -> Self::Scalar {
+                let ($($name,)+) = self;
+
+                match index {
+                    $($index => *$name,)+
+                    _ => unreachable!(),
```

Comment:
little nit: specific error text would be helpful in case we ever hit it.

```suggestion
                    _ => unreachable!("index {} out of bounds for tuple", index),
```